pip/backend.py

```diff
@@ -135,8 +134,14 @@ def _base_setup_args(self, isolate=False):
         ] + list(self.global_options)
 
     def _install_build_reqs(self, reqs, prefix):
+        # Circular import still required
```

Comment:
@pradyunsg In terms of design, this needs to be called before egg_info, which implies from InstallRequirement. But then we need a circular dependency and an instance of finder, which I'll need to track down. Do you have a better way to do this because this seems like some nasty entanglement here.